internal/controllers/user_devices_controller.go

```diff
@@ -1334,15 +1334,21 @@ func (udc *UserDevicesController) GetMintDevice(c *fiber.Ctx) error {
 		},
 	}
 
-	deviceMake := dd.Make.Name
-	deviceModel := dd.Type.Model
-	deviceYear := strconv.Itoa(int(dd.Type.Year))
+	if udc.Settings.IsProduction() {
+		mvs := registry.MintVehicleSign{
+			ManufacturerNode: makeTokenID,
+			Owner:            common.HexToAddress(*user.EthereumAddress),
+			Attributes:       []string{"Make", "Model", "Year"},
+			Infos:            []string{dd.Make.Name, dd.Type.Model, strconv.Itoa(int(dd.Type.Year))},
+		}
 
-	mvs := registry.MintVehicleSign{
-		ManufacturerNode: makeTokenID,
-		Owner:            common.HexToAddress(*user.EthereumAddress),
-		Attributes:       []string{"Make", "Model", "Year"},
-		Infos:            []string{deviceMake, deviceModel, deviceYear},
+		return c.JSON(client.GetPayload(&mvs))
+	}
+
+	mvs := registry.MintVehicleWithDeviceDefinitionSign{
+		ManufacturerNode:   makeTokenID,
+		Owner:              common.HexToAddress(*user.EthereumAddress),
+		DeviceDefinitionID: fmt.Sprintf("%s_%s_%d", dd.Make.Name, dd.Type.Model, dd.Type.Year),
```

Comment:
I think you need to get it from the DD API. It might already be lying around in that `dd` object. But I know it's more than this.

Reply:
Here: https://github.com/DIMO-Network/device-definitions-api/blob/50da7d7c322c3b183d35295c6f1cd678ddcfac5b/pkg/grpc/device_definition.proto#L40

Reply:
Maybe we should also check that this is non-empty? A little worried about that.